templates/ClientMethodOptionsSnippet.stencil

```diff
@@ -0,0 +1,23 @@
+/// User-configurable options for the `{{ model.clientName }}.{{ model.operationName }}` operation.
+public struct {{ model.name }}: AzureOptions {
+
+{% for property in model.properties %}
+    {{ property.comment }}
+    public let {{ property.name }}: {{ property.type }}{{property.defaultValue}}
+{% endfor %}
+
+    /// Initialize a `{{ model.name }}` structure.
+    /// - Parameters:
+{% for property in model.properties %}
+    ///   - {{ property.name }}: {{ property.comment }}
+{% endfor %}
+    public init(
+    {% for property in model.properties %}
+        {{ property.name }}: {{ property.type }} = nil,
```

Comment:
do we need to check if the type is an optional type before setting it to "nil' as default value?

Reply:
Possibly. A "required option" seems like an oxymoron, but I could see scenarios where it might make sense. The entire options object is optional, which implies that even if there was a "required option", if you aren't required to provide the options, then it doesn't much matter. That's why the defaults are all nil in the options struct.

Reply:
I'd say let's leave it for now and see how that plays out. I suspect that if there is a "required option" then the service will return an error. I will also ask some autorest folks about this scenario.